src/analyze/expr.rs

```diff
@@ -321,6 +337,11 @@ impl PureAnalyzer {
     fn mul(&mut self, left: Expr, right: Expr, op: BinaryOp) -> Expr {
         let location = left.location.merge(right.location);
 
+        // TODO: is this the right behavior?
+        if left.ctype == Type::Error || right.ctype == Type::Error {
+            return left;
+        }
```

Comment:
If any error is emitted, `swcc` won't emit a binary, so the returned expression is just for error recovery. I do think we should return the expression that was an error though, to avoid cascading errors.
```suggestion
        if left.ctype == Type::Error {
            return left;
        } else if right.ctype == Type::Error {
            return right;
        }
```